docs/madr/decisions/031-locality-aware-load-balancing.md

```diff
@@ -0,0 +1,281 @@
+            - ["zone-3", "zone-4"]
+          - groups:
+            - ["zone-5"]
+        fallback: any # (5)
+```
+
+
+(1) In `inZone` section, you configure the priorities of tags used for load balancing.
+
+(2) In `crossZone` section, you configure zones priority.
+
+(3) In `rules` section, you configure list of zone groups priorities, first groups will have the highest load balancing priority and last in order will have the lowest priority.
+
+(4) In `groups` section, you configure logical zone groups. Let's assume `zone-1` and `zone-2` are located in the same datacenter, so you want to load balance equally between them. `Zone-5` is located in another datacenter so it should have lower priority, that is why it is in the second groups entry.
+
+(5) At last we should apply some fallback if no dataplanes from previous configuration are available, at this moment we allow `fail` and `any` fallback
+
+### API examples based on use cases
+
+#### Keeping traffic in same zone (disable cross zone traffic)
+
+![Use case 1](assets/031/use_case_1.png)
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAware:
+      inZone: []
+```
+
+#### Load balancing based on priorities within a zone. Example: node → AZ → region
+
+![Use case 2](assets/031/use_case_2.png)
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAware:
+      inZone: ["k8s.io/node", "k8s.io/az"]
+```
+
+#### Prefer traffic in same zone with fallback to any other zone
+
+![Use case 3](assets/031/use_case_3.png)
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAware:
+        inZone: []
+    crossZone:
+      fallback: any
+```
+
+#### Load balance equally to list of zones
+
+![Use case 4](assets/031/use_case_4.png)
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAware:
+      inZone: []
+    crossZone:
+      - groups:
+        - ["zone-1", "zone-3"]
+```
+
+#### When leaving zone, target specific zones first
+
+![Use case 5](assets/031/use_case_5.png)
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAware:
+      inZone: []
+    crossZone:
+      - groups:
+        - ["zone-1"]
+      - groups:
+        - ["zone-3"]
+```
+
+#### Exclude single zone from load balancing
+
+![Use case 6](assets/031/use_case_6.png)
+
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults:
+    localityAware:
+      inZone: []
+    crossZone:
+      - groups:
+        - ["zone-1"]
+      fallback: fail
+```
+
+TODO What if i have lots of zones and don’t want to write all of them during the incident. I would like to just exclude single zone, do we want to add something like:
```

Comment:
You can still set Zone#enabled=false to stop syncing ZoneIngress of this zones to other zones. Therefore excluding the cross zone traffic to this specific zone

Reply:
But that's for the entire zone and has side effects no? I find being able to exclude 1 zone for 1 service sounds feasible